sharding/sharding.go

```diff
@@ -149,15 +152,6 @@ func (r *ShardingFerry) Run() {
 	r.Ferry.WaitUntilBinlogStreamerCatchesUp()
 
 	var err error
-
-	if r.config.RunFerryFromReplica {
-		err := r.initializeWaitUntilReplicaIsCaughtUpToMasterConnection()
-		if err != nil {
-			r.logger.WithField("error", err).Errorf("could not open connection to master replica")
-			r.Ferry.ErrorHandler.Fatal("sharding", err)
-		}
-	}
```

Comment:
The reason this was here is that we pt-kill idle connections on the db. So, if we open the connection too soon it might get killed by the server by the time we need it.

Reply:
I thought the Go driver reconnect automatically?